gql/dsl.py

```diff
@@ -1,179 +1,378 @@
+import logging
 from collections.abc import Iterable
-from functools import partial
+from typing import Any, Callable, Dict, List, Optional, Union, cast
 
 from graphql import (
     ArgumentNode,
     DocumentNode,
     EnumValueNode,
     FieldNode,
     GraphQLEnumType,
+    GraphQLField,
     GraphQLInputField,
     GraphQLInputObjectType,
+    GraphQLInputType,
+    GraphQLInterfaceType,
     GraphQLList,
+    GraphQLNamedType,
     GraphQLNonNull,
+    GraphQLObjectType,
+    GraphQLScalarType,
+    GraphQLSchema,
     ListValueNode,
     NameNode,
     ObjectFieldNode,
     ObjectValueNode,
     OperationDefinitionNode,
     OperationType,
     SelectionSetNode,
+    ValueNode,
     ast_from_value,
     print_ast,
 )
 from graphql.pyutils import FrozenList
 
 from .utils import to_camel_case
 
+log = logging.getLogger(__name__)
 
-class DSLSchema(object):
-    def __init__(self, client):
-        self.client = client
+GraphQLTypeWithFields = Union[GraphQLObjectType, GraphQLInterfaceType]
```

Comment:
I would not create an alias for this, but use it inline. Also not that input types also have fields, so the name is misleading.

Reply:
Indeed, it is better. Done